### PR TITLE
ATEAM-266: Move build examples script out of gulpfile

### PR DESCRIPTION
**Pull request by jakejrichards**
This PR moves the `buildExamples` function out of gulpfile and into its own script file that can be invoked via `npm run examples`. I also needed to switch to a vanilla js implementation, which isn't too bad. I pulled in `fs-extra` for the added functionality to make it easier to implement, and since I did that I went ahead and dropped `fs-then-native` since `fs-extra` has all the same features and more.

### CODE REVIEW

File: scripts/build-examples.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        const res = path.resolve(dir, subdir);
+        return (await fs.stat(res)).isDirectory() ? getFiles(res) : res;
+    }));
+    return files.reduce((a, f) => a.concat(f), []);
+}
+
+async function buildExamples(destPath = DEST_DIR, sdkUrl = null) {
+    await fs.copy('examples', destPath);
+    const files = await getFiles(destPath);
+    for (let file of files) {
```
Comment (justinaray) [inline scripts/build-examples.js:27]: I dont think we're guaranteed of a clean dist directory here.  there might be other build artifacts that could match and get corrupted.  I wonder if we should use a temp directory to mutate and then copy into dest_dir?  Or, we could look into streaming options?

File: package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "lint": "eslint gulpfile.js src/*.js src/**/*.js",
     "build": "gulp",
     "docs": "node scripts/build-docs.js",
+    "examples": "node scripts/build-examples.js",
```
Comment (justinaray) [inline package.json:23]: Do we need a separate build script for the examples?  Might just be part of `start` `start:watch`.  I don't have a strong opinion here.  Just raising it as a thought.

File: gulpfile.js [hunk 3]
```diff
@@ -173,8 +167,8 @@ gulp.task('watch', function () {
 });
 
 gulp.task('serve', ['clean'], function () {
-    let buildSdkForDevServer = buildStdDists.bind(this, path.resolve(DEST_DIR, 'vendor'));
-    let buildExamplesForDevServer = buildExamples.bind(this, DEST_DIR, `vendor/${BROWSER_FILENAME}`);
+    let buildSdkForDevServer = () => buildStdDists(path.resolve(DEST_DIR, 'vendor'));
```
Comment (justinaray) [inline gulpfile.js:170]: No more bind for scope, yay!

File: scripts/build-examples.js (hunk 1 above)
Comment (jakejrichards) [reply scripts/build-examples.js:27]: Yeah, that's a fair point. I'll look into this!

File: package.json (hunk 2 above)
Comment (jakejrichards) [reply package.json:23]: Yeah... reflecting on it, building the examples without first running the start script wouldn't make much sense.

Comment (justinaray) [reply package.json:23]: I think we should remove this script then since you removed the cli support from the script.

Comment (jakejrichards) [reply package.json:23]: Oh, that's strange I could've sworn I removed it with my most recent commit. I'll fix it up, good catch!

**Pull request merged**